### PR TITLE
feat: Ability to remove unused imports from multi-use statements

### PR DESCRIPTION
**Pull request by Wirone**
Built on top of #7814 and that PR must be merged before.

This utilises chunk-based `NamespaceUseAnalysis` to remove parts of the multi-use statements and can also clean whole statement if every chunk was removed.

### CODE REVIEW

File: tests/Fixer/Import/NoUnusedImportsFixerTest.php [hunk 1]
```diff
@@ -1319,42 +1319,80 @@ public function foo() {}
...
-    fn_a(ClassA::test, new C());
-',
+        yield 'grouped imports' => [
+            <<<EOF
+                <?php
+                use some\\y\\{ClassA, ClassC as C,};
+                use function some\\a\\{
+                    {$empty}
+                    fn_b,
+                    {$empty}
```
Comment (Wirone) [inline tests/Fixer/Import/NoUnusedImportsFixerTest.php:1352]: This is last failing test case, I found it problematic and wanted to discuss it. First of all: should we expect fixer to clean up empty lines? If yes, I'll try to find a good way (there are edge cases, because if there's a comment the line should be kept as-is). If not, can anyone point me to good example how to handle trailing whitespace in string? There's probably also a issue with merging surrounding whitespace during clean-up. Anyway, it's last(?) thing to do here.

Comment (Wirone) [reply tests/Fixer/Import/NoUnusedImportsFixerTest.php]: Fixed in [1cc34191](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7815/commits/1cc34191d1fd3b9dc45c4441341760b081226ebd).

**Review by kubawerlos — APPROVED** [review]

File: src/Fixer/Import/NoUnusedImportsFixer.php [hunk 2]
```diff
@@ -261,30 +366,81 @@ private function removeUseDeclaration(Tokens $tokens, NamespaceUseAnalysis $useD
...
+     *
+     * @param -1|1 $direction
+     *
+     * @return array{0: int, 1: bool}
+     */
+    private function scanForNonEmptyTokensUntilNewLineFound(Tokens $tokens, int $index, int $direction): array
+    {
+        $hasNonEmptyToken = false;
+
+        while (\is_int($index) && $index > 0) {
```
[the remark this answers is not in the document]
Comment (Wirone) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: I've refactored this method a little, should be better now 🙂.

[the remark this answers is not in the document]
Comment (julienfalque) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: Mmh now the method might return `null` as the first value of the tuple, but usages don't check that.

[the remark this answers is not in the document]
Comment (Wirone) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: The tuple is used in 2 places, but only one is using information about new line's token index (under `0` key). I've improved the second one, which is using this info, even though it was working correctly (see additional test cases, I didn't have to fix anything to get them working). Explanation is in [commit's message](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7815/commits/99aa87bf537ee07cfd130197698ba59106db261d).

**Review by julienfalque — APPROVED** [review]

File: src/Fixer/Import/NoUnusedImportsFixer.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (julienfalque) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: I find it weird that PHPStan never complained.
